packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2158,7 +2256,15 @@ export class ContainerRuntime
 		// Do shallow copy of message, as the processing flow will modify it.
 		const messageCopy = { ...messageArg };
 		for (const message of this.remoteMessageProcessor.process(messageCopy)) {
-			this.processCore(message, local, modernRuntimeMessage);
+			if (modernRuntimeMessage) {
+				this.processCore({
+					message: message as InboundSequencedContainerRuntimeMessage,
+					local,
+					modernRuntimeMessage,
+				});
+			} else {
```

Comment:
I don't think we fix it in this PR, but one thing that has been bugging me is that there isn't a strict assertion/typing/rule that the message.type for all of the messages that come out of remoteMessageProcessor.process must be MessageType.Operation or not just as messageArg.type is. Seems fragile.

Reply:
If I understand what you're saying, it doesn't actually hold - Rather, if `messageArg.type` is `MessageType.Operation`, then the output will be "unpacked" such that each `message.type` will be `ContainerMessageType` (or unknown in the future case).  Otherwise, _I believe_ the output will be a singleton array containing `messageCopy`.

Reply:
And yeah, I'd like to confirm and codify that as well.

Reply:
Adding comment.